packages/dds/tree/src/test/time/tableTree.spec.ts

```diff
@@ -0,0 +1,646 @@
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function runBenchmark({
+	title,
+	tableSize,
+	cellValue,
+	operation,
+	minBatchDurationSeconds = 0,
+	maxBenchmarkDurationSeconds,
+}: {
+	title: string;
+	tableSize: number;
+	cellValue: string;
+	operation: (tree: TableTreeDefinition) => void;
+	minBatchDurationSeconds?: number;
+	maxBenchmarkDurationSeconds: number;
+}) {
+	benchmark({
+		type: BenchmarkType.Measurement,
+		title,
+		benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+			let duration: number;
+			do {
+				assert.equal(state.iterationsPerBatch, 1, "Expected exactly one iteration per batch");
+
+				// Setup
+				const localTree: TableTreeDefinition = createTableTree(tableSize, cellValue);
+
+				// Operation
+				const before = state.timer.now();
+				operation(localTree);
+				const after = state.timer.now();
+
+				// Measure
+				duration = state.timer.toSeconds(before, after);
+			} while (state.recordBatch(duration));
+		},
+		minBatchDurationSeconds,
+		maxBenchmarkDurationSeconds,
+	});
+}
+
+/**
+ * This function runs a benchmark for undo/redo operations on a SharedTree.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function runUndoRedoBenchmark({
+	title,
+	tableSize,
+	cellValue,
+	setupOperation,
+	stackOperation,
+	minBatchDurationSeconds = 0,
+	maxBenchmarkDurationSeconds,
+}: {
+	title: string;
+	tableSize: number;
+	cellValue: string;
+	/**
+	 * A function that sets up the operation to be performed on the tree.
+	 */
+	setupOperation: (tree: TableTreeDefinition) => void;
+	/**
+	 * The operation to perform on the stack. This should be a function that takes an UndoRedoStackManager
+	 * and performs the desired operation.
+	 */
+	stackOperation: (tree: TableTreeDefinition) => void;
+	minBatchDurationSeconds?: number;
+	maxBenchmarkDurationSeconds: number;
+}) {
+	benchmark({
+		type: BenchmarkType.Measurement,
+		title,
+		benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+			let duration: number;
+			do {
+				assert.equal(state.iterationsPerBatch, 1, "Expected exactly one iteration per batch");
+
+				// Setup
+				const localTree: TableTreeDefinition = createTableTree(tableSize, cellValue);
+				setupOperation(localTree);
+
+				// Operation
+				const before = state.timer.now();
+				stackOperation(localTree);
+				const after = state.timer.now();
+
+				// Measure
+				duration = state.timer.toSeconds(before, after);
+
+				// Cleanup
+				localTree.unsubscribe();
+			} while (state.recordBatch(duration));
+		},
+		minBatchDurationSeconds,
+		maxBenchmarkDurationSeconds,
+	});
+}
+
+describe("SharedTree execution time", () => {
+	// The value to be set in the cells of the tree.
+	const cellValue = "cellValue";
+	// The test tree's size will be 10*10, 100*100.
+	// Tree size 1000 benchmarks removed due to high overhead and unreliable results.
+	const treeSizes = isInPerformanceTestingMode
```

Comment:
We should probably call this `tableSize` (or something) instead. This doesn't describe the size of the tree, it describes the size of the table modeled by the tree.